kolibri/core/logger/api.py

```diff
@@ -55,6 +58,35 @@ def filter_learner_group(self, queryset, name, value):
             target_user=F("user"),
         )
 
+class LoggerViewSet(viewsets.ModelViewSet):
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', False)
+        model = self.queryset.model
+        lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+        try:
+            instance = model.objects.get(id=self.kwargs[lookup_url_kwarg])
```

Comment:
One of the really important things that `get_object` does is check the appropriate permissions for the current user against the object being updated:

https://github.com/encode/django-rest-framework/blob/master/rest_framework/generics.py#L101

I'm guessing that's where a good portion of the savings may be coming from?

If the permission check is slowing things down, we can do some optimizations there. We have a very general-purpose permission system that allows for complex relationships between users and objects to be defined. See:
https://kolibri-dev.readthedocs.io/en/develop/architecture/uap/concepts_and_definitions.html
and
https://kolibri-dev.readthedocs.io/en/develop/architecture/uap/implementation.html

But we can look into "short-circuiting" it in this common case (if it isn't already) where the log is directly associated with the current user.

Reply:
Hi, Jamie, good catch. I hadn't seen the permissions checks. I have added it to this PR.
Obviously, this reduces the performance improvement (this check is making one sql query, I think it could be cached instead of looking to the db for the same permission in every request, but that's another story).
Anyway, you can see in the chart below that that's not where most of the time is spent, the `get_object_or_404` function takes longer because it makes a query for each of the fields of the object, instead of fetching the whole object in one query.
![has_permissions](https://user-images.githubusercontent.com/1008178/42393227-f8ff10fe-8155-11e8-9b89-64cc1ee31527.png)